MdeModulePkg/Core/Dxe/Misc/MemoryProtectionSupport.c

```diff
@@ -1301,15 +1301,16 @@ FillInMemoryMap (
                           &OverlapLength,
                           &GcdType
                           );
+      if (GcdType != EfiGcdMemoryTypeNonExistent) {
+        POPULATE_MEMORY_DESCRIPTOR_ENTRY (
+          NewMemoryMapCurrent,
+          StartOfAddressSpace,
+          EfiSizeToPages (OldMemoryMapCurrent->PhysicalStart - StartOfAddressSpace - RemainingLength),
+          GcdTypeToEfiType (&GcdType)
+          );
+        NewMemoryMapCurrent = NEXT_MEMORY_DESCRIPTOR (NewMemoryMapCurrent, *DescriptorSize);
+      }

```

Comment:
To clarify, in the case of holes in the memory map that get remapped later, we would not map them in this first pass (as they are EfiGcdMemoryTypeNonExistent) but when the memory gets added, later, we will simply insert the memory region into the proper place in the memory map? I.e. this won't be an updating of region type in the memory map, but actually a physical addition of a region to the map.

Reply:
This code is running as part of CpuDxe.

CpuDxe is operating on a copy of the memory map, as it exists at this point in execution.
The local copy of the memory map will be used to update the cpu page tables.

When PciHostBridge runs, https://github.com/microsoft/mu_basecore/blob/release/202208/MdeModulePkg/Bus/Pci/PciHostBridgeDxe/PciHostBridge.c#L545
it should add memory space, set attributes, and allocate memory space, which should follow mu memory protections, as reported for the hob.